### PR TITLE
Added order status management endpoints in the FastAPI application. I…

### PR DESCRIPTION
**Pull request by sajalahalih**
…mplemented CRUD operations for managing order statuses, including:

- `POST /statuses/` to create a new status with validation for unique names.
- `GET /statuses/{status_id}` to retrieve details of a specific status by its ID.
- `PUT /statuses/{status_id}` to update an existing status name with uniqueness validation.
- `DELETE /statuses/{status_id}` to remove a status, ensuring it is not in use by any orders.

Enhanced code documentation with detailed comments for clarity and maintainability.

### CODE REVIEW

**Review by FatmaQunnies24 — APPROVED** [review]

File: app/__pycache__/__init__.cpython-312.pyc
Comment (Mohammed-samaheen) [inline app/__pycache__/__init__.cpython-312.pyc:1]: Please remove all the` __pycache__ `files from this PR.

File: app/api/routes/OrderStatus.py
Comment (Mohammed-samaheen) [inline app/api/routes/OrderStatus.py:1]: Please use snake_case for file naming.

File: app/api/routes/OrderStatus.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    Returns:
+    - The newly created OrderStatus object.
+    """
+    existing_status = await OrderStatus.get(name=name)
+    if existing_status:
+        raise HTTPException(status_code=400, detail="Status name must be unique")
+
+    # Create and save the new status
+    new_status = OrderStatus(name=name, created_at=datetime.utcnow())
+    await new_status.save()
```
Comment (Mohammed-samaheen) [inline app/api/routes/OrderStatus.py:31]: Did you test your changes? I couldn't find this syntax in the SQLModel documentation. Here's an [example](https://sqlmodel.tiangolo.com/tutorial/insert/#create-a-model-instance) of how it's done.

Comment (Mohammed-samaheen) [reply app/api/routes/OrderStatus.py:31]: If such syntax exists, please send me a link

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]
Just a small change; overall, nice job!

**Pull request closed without merging**